lightly/api/api_workflow_predictions.py

```diff
@@ -0,0 +1,152 @@
+from concurrent.futures import ThreadPoolExecutor
+from typing import Dict, List, Union, Optional
+
+import tqdm
+
+from lightly.api.prediction_singletons import PredictionSingletonClassificationRepr
+from lightly.api.utils import retry
+from lightly.openapi_generated.swagger_client import (
+    PredictionTaskSchema,
+    SamplePartialMode,
+    PredictionSingletonClassification,
+    PredictionSingletonObjectDetection,
+    PredictionSingletonInstanceSegmentation,
+    PredictionSingletonKeypointDetection,
+)
+
+
+class _PredictionsMixin:
+    def create_or_update_prediction_task_schema(
+        self,
+        schema: PredictionTaskSchema,
+        prediction_uuid_timestamp: int,
+    ) -> None:
+        """Creates or updates the prediction task schema
+
+        Args:
+            schema:
+                The prediction task schema.
+            prediction_uuid_timestamp:
+                This timestamp is used as a key to distinguish different predictions for the same sample.
+                Get it e.g. via 'int(time.time())'
```

Comment:
This is the parameter as defined in the API. See https://github.com/lightly-ai/lightly/pull/985#discussion_r1029522176
If you want to challenge the current design of it in the API, I would do that outside of this PR.

Reply:
Sound good to have it out of scope of this PR (then we will need to update two places though).

@japrescott Would you know the answer to my question?

Reply:
See https://linear.app/lightly/issue/LIG-1303/upload-predictions-to-api

Reply:
@michal-lightly I see your point that its "counter intuitive". The reason its a mix is because I didn't want to create more endpoints for versioning etc. before we don't even know if the basic concept works :)

A customer has predictions at time X. Then uses AL, labels them, retrains models, gets new predictions. Then uses Lightly again at time Y. The idea was that the customer can see the difference between time X and time Y in the UI.

"Correct" would have been that the customer could define a name and we would then reference that entity which would have a timestamp attached for sorting purposes. I decided for the "less complicated" way but maybe that is not ideal.

> What if we want to upload two different prediction schemas and it happens to be in the same second?

A schema is unique by `datasetId+uuidtimestamp+name` so this does not matter.

> Why not just use an incremental uuid

Sure, that is up to you. We can also roll with `0` for the moment and always overwrite the schemas. Or we can always set it to the current timestamp so there is a "time element" associated to the value.

Reply:
Jeremy and I had a discussion in which Jeremy explained the meaning of the field. It is worth knowing that:
* This interface was designed 9 months ago, and is imperfect
* Changing it now (even a rename) would not be trivial

We decided to keep the API as is, but give it a very good documentation.

The `prediction_uuid_timestamp` should be properly named `predicionVersionId`. It is any id specified by the user. If the user does not care about keeping history, they can simply always pass the same number.

Note that originally the endpoint was not meant for customer use. The idea was to display a timestamp in the UI and the field was (a bit hackily) meant to double as the timestamp.

Reply:
For the ApiWorkflowClient function, I propose to change the parameters as follows:
* Call the param `predicion_version_id` and make it optional
* If not specified, pass always `-1` to the api

Pros:
* Clearer to understand
* Easier to use
Cons:
* Not 1:1 correspondence to the API call
* The version id is an int, but str would be more natural
* Special value -1

Reply:
I agree and will change it.